pkg/es/client/index_client.go

```diff
@@ -249,6 +249,24 @@ func (i IndicesClient) CreateTemplate(template, name string) error {
 	return nil
 }
 
+// Create Template for ESV8
```

Comment:
as far as I can see, the only difference in this method is the endpoint `_index_template` instead of `_template` in the function above. I think this is not a good reason to have a full new function declared at the interface level. Can we not just pass the version to the IndicesClient and make the endpoint name conditional on that version?